src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1293,21 +1301,17 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
 
                             var factory = new SyntheticBoundNodeFactory(methodSymbol, syntax, compilationState, diagsForCurrentMethod);
 
-                            // Iterators handled in IteratorRewriter.cs
-                            if (!methodSymbol.IsIterator)
-                            {
-                                var boundStatementsWithNullCheck = LocalRewriter.TryConstructNullCheckedStatementList(methodSymbol.Parameters, boundStatements, factory);
+                            var nullCheckStatements = LocalRewriter.TryConstructNullCheckedStatementList(methodSymbol.Parameters, factory);
 
-                                if (!boundStatementsWithNullCheck.IsDefault)
+                            if (!nullCheckStatements.IsDefault)
+                            {
+                                boundStatements = nullCheckStatements.Concat(boundStatements);
+                                hasErrors = nullCheckStatements.HasErrors() || diagsForCurrentMethod.HasAnyErrors();
+                                SetGlobalErrorIfTrue(hasErrors);
+                                if (hasErrors)
                                 {
-                                    boundStatements = boundStatementsWithNullCheck;
-                                    hasErrors = boundStatementsWithNullCheck.HasErrors() || diagsForCurrentMethod.HasAnyErrors();
-                                    SetGlobalErrorIfTrue(hasErrors);
-                                    if (hasErrors)
-                                    {
-                                        _diagnostics.AddRange(diagsForCurrentMethod);
-                                        return;
-                                    }
+                                    _diagnostics.AddRange(diagsForCurrentMethod);
+                                    return;
                                 }
```

Comment:
It seems surprising we're calling `SetGlobalErrorIfTrue()` here and returning early only if there were null checks. What if `diagsForCurrentMethod.HasAnyErrors()` was `true` even without null checks?

Reply:
I think the `if (hasErrors)` block is redundant here.